### ISSUE TITLE
Allow local doh for dnscrypt 

### ISSUE THREAD

**Issue by mvevitsis** [Gedsh/InviZible#275]
App will not allow dnscrypt to start if local_doh settings in toml are uncommented. 

Otherwise, I think theoretically it should work if the user certificate is installed to the android system.

**Comment by Gedsh** [comment]
This feature is not currently supported. InviZible does not include the required certificate file. I will implement it over time if you are interested in having it available.

**Comment by mvevitsis** [comment]
I want to use my own certificate file. 
I already have one I made with mkcert. 
However, even with correct the file path specified, I cannot get inviZible to start dnscrypt-proxy with the lines uncommented.

**Comment by Gedsh** [comment]
> with correct the file path to the certificates specified

You should put the certificate in an app's internal folder. For example, /data/data/pan.alexander.tordnscrypt/app_data/dnscrypt-proxy, which can only be done with the root.

**Comment by mvevitsis** [comment]
I don't have root, so you would need to add the all file access permission so I can store them somewhere else.

**Comment by Gedsh** [comment]
This would require permission to access all files, which users won't like. In addition, it violates Play Market policy. The only solution is to implement a way to add the certificate to an internal folder. This is possible, but requires time and effort.

**Comment by mvevitsis** [comment]
> This would require permission to access all files, which users won't like. In addition, it violates Play Market policy. The only solution is to implement a way to add the certificate to an internal folder. This is possible, but requires time and effort.

Ok, well if possible I would like to see it implemented.

**Comment by Gedsh** [comment]
> pushed my certs using ADB

This folder is not accessible in modern android versions through the regular file manager. I don't think most users will like using ADB.

**Comment by mvevitsis** [comment]
> > pushed my certs using ADB
> 
> This folder is not accessible in modern android versions through the regular file manager. I don't think most users will like using ADB.

Right, you cannot access this folder without adb (or shizuku/rish if you want to do it on your phone).

**Comment by Suznyan** [comment]
> There is no need for root.
> 
> Your app already has a folder in Android/data to store some files I made a new folder in there called certificates and pushed my certs using ADB. Specified these in the toml and it works:
> 
> ![Screenshot_20240802-070659_Chrome.png](https://github.com/user-attachments/assets/31441bec-e1e6-4acc-b66b-bb143bf2ac55)
> 
> Success.
> 
> Notes for anyone else who needs this: Install root ca using Android system settings. Chrome will recognize it automatically but for Firefox you need to activate developer mode (press Firefox logo a bunch of times in 'about') then go to secret settings and enable allow 3rd party ca certs.

Hello, may I ask what your toml for the cert path is like?

**Comment by mvevitsis** [comment]
> > There is no need for root.
> > 
> > Your app already has a folder in Android/data to store some files I made a new folder in there called certificates and pushed my certs using ADB. Specified these in the toml and it works:
> > 
> > ![Screenshot_20240802-070659_Chrome.png](https://github.com/user-attachments/assets/31441bec-e1e6-4acc-b66b-bb143bf2ac55)
> > 
> > Success.
> > 
> > Notes for anyone else who needs this: Install root ca using Android system settings. Chrome will recognize it automatically but for Firefox you need to activate developer mode (press Firefox logo a bunch of times in 'about') then go to secret settings and enable allow 3rd party ca certs.
> 
> Hello, may I ask what your toml for the cert path is like?


cert_file = '/storage/emulated/0/Android/data/pan.alexander.tordnscrypt.stable/certificates/nameofyourfile.pem

Same for the key, just specify the other file name.
